### PR TITLE
Feature/nurse connect profiles

### PR DESCRIPTION
**Pull request by phala**

### CODE REVIEW

File: nurseconnect/forms.py [hunk 1]
```diff
@@ -4,18 +4,19 @@
...
 from django.utils.translation import ugettext_lazy as _
 
-from phonenumber_field.formfields import PhoneNumberField
-from phonenumber_field.validators import validate_international_phonenumber
 from wagtail.contrib.settings.context_processors import SettingsProxy
 from wagtail.wagtailcore.models import Site
 
+from nurseconnect.formfields import PhoneNumberField
+
 ZATEL_REG = r"^((?:\+27|27)|0)[\s-]?(\d{2})[\s-]?(\d{3})[\s-]?(\d{4})[\s]*$"
```
Comment (chrislombaard) [inline nurseconnect/forms.py]: I think this is unused now, so please remove.

File: nurseconnect/forms.py [hunk 2]
```diff
@@ -24,7 +25,26 @@ class RegistrationForm(forms.Form):
...
             }
         ),
-        label=_("Mobile Number")
+        label=_("Mobile Number"),
+        error_messages={
+            "invalid": "Please enter a valid South African cellphone number."
+        },
+    )
+
+    clinic_code = forms.RegexField(
```
Comment (chrislombaard) [inline nurseconnect/forms.py:33]: `max_length`, `min_length`?

File: nurseconnect/forms.py [hunk 3]
```diff
@@ -180,22 +203,10 @@ def __init__(self, *args, **kwargs):
...
-                self.cleaned_data["username"] = raw_username
-
-        if self.user.username != self.cleaned_data["username"]:
-            if User.objects.filter(
-                username__iexact=self.cleaned_data["username"]
-            ).exists():
-                self._errors["username"] = self.error_class(
-                    ["Username already exists."]
-                )
+        if User.objects.filter(
```
Comment (chrislombaard) [inline nurseconnect/forms.py]: Does the current implementation of PhoneNumberField normalize the phonenumber input to +27?

File: nurseconnect/models.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+class UserProfile(MoloUserProfile):
+    """
+    NurseConnect requires a few modifications to the standard way Profiles are
+    handled by molo.profiles. This model serves to implement these.
+    """
+    clinic_code = models.CharField(
+        max_length=6,
+        blank=True,
+        null=True)
```
Comment (chrislombaard) [inline nurseconnect/models.py]: Bracket on the next line please.

**Review by chrislombaard — CHANGES_REQUESTED** [review]

File: nurseconnect/forms.py (hunk 2 above)
Comment (phala) [reply nurseconnect/forms.py:33]: No need for those two. The regex pattern enforces it anyway. Having them would result in tow errors reported for the same thing.

**Review by chrislombaard — APPROVED** [review]
+1

**Pull request merged**